src/sentry/api/helpers/group_index.py

```diff
@@ -319,3 +334,558 @@ def delete_groups(request, project, group_list, delete_type):
             user=request.user,
             delete_type=delete_type,
             sender=delete_groups)
+
+
+def self_subscribe_and_assign_issue(acting_user, group):
+    # Used during issue resolution to assign to acting user
+    # returns None if the user didn't elect to self assign on resolution
+    # or the group is assigned already, otherwise returns Actor
+    # representation of current user
+    if acting_user:
+        GroupSubscription.objects.subscribe(
+            user=acting_user,
+            group=group,
+            reason=GroupSubscriptionReason.status_change,
+        )
+        self_assign_issue = UserOption.objects.get_value(
+            user=acting_user, key='self_assign_issue', default='0'
+        )
+        if self_assign_issue == '1' and not group.assignee_set.exists():
+            return Actor(type=User, id=acting_user.id)
+
+
+def update_groups(request, projects, organization_id, search_fn):
```

Comment:
projects here would be the projects a user has access to within the org. I will probably add something within this function that filters that down to the projects that actually belong to the groups we're updating as I think that will help the efficiency in some cases. Will do that before adding to the org endpoint but in a separate diff than this.